### ISSUE TITLE
New name for `wgpu` rust library

### ISSUE THREAD

**Issue by Gordon-F** [gfx-rs/wgpu#2534]
`wgpu` is a great library, but I think its name prevents it from becoming even more popular.

Why is it necessary to rename the library:
1. People don't understand the difference between `wgpu` (rust-lib), `wgpu-core` (WebGPU implementation), `wgpu-native` (c bindings) and `WebGPU` (spec)
2. People think that `wgpu` targets only for Web (I think because of `w` in name :smile:)
3. It's hard to find library source code in the repository if you are not familiar with it (`wgpu/wgpu` folder? 😕)

`Naga` is a good example of naming. It's not a `wgpu-shader-translation`. It has its own catchy name.

**Comment by kvark** [comment]
Honestly, I think this isn't worth fighting for. We choose the current naming scheme, and we should just stick with it. The cost of renaming overweights a possible confusion by the users at this point. The biggest challenge for the library is to become revitalized by the community, given my reduced attention - let's focus on this and not be distracted.

**Comment by cwfitzgerald** [comment]
I agree with kvark here, I think this ship has long sailed. Renaming would increase the amount of people finding the library, but changing it would cause a massive inconvenience to all our existing users and likely hemorrhage more than we gained. It would also increase confusion, especially considering the already confusing history of the gfx-rs project. 

I think the solution is to have clear and consistent messaging about what exactly we are, so people can get their own questions answered. Both our repo description and the readme intentionally bury the lede of us being a WebGPU implementation for this very reason. 

Source code visibility is a separate issue, and I think could be rectified by reordering the readme a bit to have the "getting started" page with links to the examples be on the first screen of text.

**Comment by cdecompilador** [comment]
Even if the name has those issues, still is I like it and has already got some spread, changing it only would create confusion at this point, when `gfx-rs` was a thing it was logical as it was not as used as `wgpu` is now

**Comment by cwfitzgerald** [comment]
Sounds like this issue is pretty settled.

**Comment by aloucks** [comment]
Interestingly, I would argue the opposite. `wgpu` is a perfect name, it's the Rust implementation of the `WebGPU` spec. The abbreviation both makes sense and is easy to type and remember. `naga` sounds more like a code word or a team name. I'm not suggesting that it be renamed, however.

> It's hard to find library source code in the repository if you are not familiar with it (wgpu/wgpu folder? 😕)

I think a good pattern is having a `crates` directory at the root of the project (this is what [bevy](https://github.com/bevyengine/bevy) does).
